http3/server.go

```diff
@@ -34,11 +37,38 @@ type requestError struct {
 	connErr   errorCode
 }
 
+func (r *requestError) HasError() bool {
+	return r.err != nil || r.streamErr != 0 || r.connErr != 0
+}
+
+func (r *requestError) String() string {
+	ret := make([]string, 0)
+	if r.err != nil {
+		ret = append(ret, r.err.Error())
+	}
+	if r.streamErr != 0 {
+		ret = append(ret, r.streamErr.String())
+	}
+	if r.connErr != 0 {
+		ret = append(ret, r.connErr.String())
+	}
+	return strings.Join(ret, ", ")
+}
+
+type ctrlStream struct {
```

Comment:
Why do you define a struct that combines the two streams?

Reply:
To make it easier to keep track of the max stream ID for each connection. I can remove it if necessary, but it make the relevant code cleaner